src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -393,6 +403,46 @@ bool GcInfoDecoder::IsInterruptible()
     return m_IsInterruptible;
 }
 
+bool GcInfoDecoder::HasInterruptibleRanges()
+{
+    _ASSERTE(m_Flags & (DECODE_INTERRUPTIBILITY | DECODE_GC_LIFETIMES));
+    return m_NumInterruptibleRanges > 0;
+}
+
+bool GcInfoDecoder::IsSafePoint()
+{
+    _ASSERTE(m_Flags & (DECODE_INTERRUPTIBILITY | DECODE_GC_LIFETIMES));
+    return m_SafePointIndex != m_NumSafePoints;
+}
+
+bool GcInfoDecoder::AreSafePointsInterruptible()
+{
+    if (m_Version < 3)
+        return false;
+
+    // zero initialized
+    static bool fInterruptibleCallSitesInited;
+    static uint64_t fInterruptibleCallSites;
+
+    if (!fInterruptibleCallSitesInited)
+    {
+#ifdef FEATURE_NATIVEAOT
+        fInterruptibleCallSites = 1;
+        g_pRhConfig->ReadConfigValue("InterruptibleCallSites", &fInterruptibleCallSites);
+#else
+        fInterruptibleCallSites = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_InterruptibleCallSites);
```

Comment:
It would be better for this config in code manager so that the decoder stays pure.

Reply:
Decoder is easily constructed without consulting code manager. For example GC stress just makes decoders as needed - https://github.com/dotnet/runtime/blob/5ce26746645a05b86beec3edb2af35f022e7867d/src/coreclr/vm/gccover.cpp#L487
 
On the other hand not all the uses will ask for safe point interruptibility. 

This was my reasoning for placing the check here. - No chance for various decoder instances to disagree and it also can be delayed until GC happens.

Reply:
Other examples: 
* exception handling creates decoders, but would not be interested in interruptibility.
https://github.com/dotnet/runtime/blob/5ce26746645a05b86beec3edb2af35f022e7867d/src/coreclr/vm/exceptionhandling.cpp#L1338

* eetwain will create decoder while figuring if method is hijackable
https://github.com/dotnet/runtime/blob/5ce26746645a05b86beec3edb2af35f022e7867d/src/coreclr/vm/eetwain.cpp#L5944

I think placing the check in something that does not need to be around when creating the decoder could be a bit fragile.

Reply:
This is only about `IsInterruptibleSafePoint` and `AreSafePointsInterruptible` properties. Is that right?

I think that the logic in these properties (including reading the config) can be moved out to code manager and the code manager / GC stress can call them on code manager. I do not see fragility in that. It is very few places.

Reply:
I see. Basically make the decoder unaware and make the callers of these properties check the knob prior to calling (since there are not many). 
Then codeman can own the check. That would work too.